cirq/ops/gate_operation.py

```diff
@@ -63,6 +63,14 @@ def with_gate(self, new_gate: 'cirq.Gate') -> 'cirq.Operation':
             return self
         return new_gate.on(*self.qubits)
 
+    def _with_measurement_key_mapping_(self, key_map: Dict[str, str]):
+        new_gate = protocols.with_measurement_key_mapping(self.gate, key_map)
+        if new_gate is NotImplemented:
+            return NotImplemented
+        if new_gate is self.gate:
```

Comment:
I think this should be `==` ?

Reply:
Oh scratch that - I see what you are doing now - if the mapping is not applicable, the measurement gate is going to be the same instance, the operation as well. I'll resolve this and let's continue the discussion on the other comment about whether a `with_*` method should return `self` ever or not